app/shop.py

```diff
@@ -0,0 +1,46 @@
+from datetime import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list,
+            prices: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.prices = prices
+        self.purchases = {}
+
+    def calculate_total_cost_of(self, customer_name: str, cart: dict) -> float:
+        receipt = {"total_cost": 0,
+                   "cart": []}
+        for product in cart:
+            price = self.prices.get(product)
+            if price is not None:
+                cost = round(price * cart[product], 2)
+                if cost % int(cost) == 0:
+                    cost = int(cost)
+
+                receipt["total_cost"] += cost
+                receipt["cart"].append((cart[product], product, cost))
+
+        self.purchases[customer_name] = receipt
+
+        return receipt["total_cost"]
+
+    def print_receipt(self, customer_name: str) -> None:
+        date = datetime(2021, 1, 4, 12, 33, 41).strftime("%d/%m/%Y %H:%M:%S")
```

Comment:
You must display current date

Reply:
Tests are made in a way that they are failed when the exact date is not given. So I will change it back...